spec/abstract/Schedule_A.thy

```diff
@@ -108,15 +108,11 @@ where
     cur_th \<leftarrow> gets cur_thread;
     sc_opt \<leftarrow> get_tcb_obj_ref tcb_sched_context cur_th;
     scp \<leftarrow> assert_opt sc_opt;
-    sc' \<leftarrow> get_sched_context scp;
+    sc' \<leftarrow> get_sched_context cur_sc;
```

Comment:
Good catch, this one. This was actually a bug, wasn’t it? The refill_max check is for cur_sc, not cur_thread.

Reply:
I don't know whether it was a bug in the C, but it was wrong in the abstract for sure.